pkg/imgpkg/bundle/bundle.go

```diff
@@ -29,24 +31,83 @@ type Bundle struct {
 	plainImg         *plainimg.PlainImage
 	imgRetriever     ctlimg.ImagesMetadata
 	imagesLockReader ImagesLockReader
+	imagesRef        map[string]ImageRef
 }
 
 func NewBundle(ref string, imagesMetadata ctlimg.ImagesMetadata) *Bundle {
 	return NewBundleWithReader(ref, imagesMetadata, &singleLayerReader{})
 }
 
 func NewBundleFromPlainImage(plainImg *plainimg.PlainImage, imagesMetadata ctlimg.ImagesMetadata) *Bundle {
-	return &Bundle{plainImg, imagesMetadata, &singleLayerReader{}}
+	return &Bundle{plainImg: plainImg, imgRetriever: imagesMetadata, imagesLockReader: &singleLayerReader{}, imagesRef: map[string]ImageRef{}}
 }
 
 func NewBundleWithReader(ref string, imagesMetadata ctlimg.ImagesMetadata, imagesLockReader ImagesLockReader) *Bundle {
-	return &Bundle{plainimg.NewPlainImage(ref, imagesMetadata), imagesMetadata, imagesLockReader}
+	return &Bundle{plainImg: plainimg.NewPlainImage(ref, imagesMetadata), imgRetriever: imagesMetadata, imagesLockReader: imagesLockReader, imagesRef: map[string]ImageRef{}}
 }
 
 func (o *Bundle) DigestRef() string { return o.plainImg.DigestRef() }
 func (o *Bundle) Repo() string      { return o.plainImg.Repo() }
 func (o *Bundle) Tag() string       { return o.plainImg.Tag() }
 
+func (o *Bundle) AddImagesRef(refs ...ImageRef) {
+	for _, imageRef := range refs {
+		o.imagesRef[imageRef.Image] = imageRef
+	}
+}
+
+func (o *Bundle) ImageRef(imageDigest string) (ImageRef, bool) {
+	ref, found := o.imagesRef[imageDigest]
+	return ref, found
+}
+
+func (o *Bundle) ImagesRef() []ImageRef {
+	var imgsRef []ImageRef
+	for _, ref := range o.imagesRef {
+		imgsRef = append(imgsRef, ref)
+	}
+	return imgsRef
+}
+
+func (o *Bundle) NoteCopy(processedImages *imageset.ProcessedImages, reg ImagesMetadataWriter, logger util.LoggerWithLevels) error {
+	locationsCfg := LocationConfig{
+		APIVersion: LocationAPIVersion,
+		Kind:       LocationKind,
+	}
+	var bundleProcessedImage imageset.ProcessedImage
+	for _, image := range processedImages.All() {
+		ref, found := o.ImageRef(image.UnprocessedImageRef.DigestRef)
+		if found {
+			locationsCfg.Images = append(locationsCfg.Images, ImageLocation{
+				Image:  ref.Image,
+				Bundle: ref.IsBundle,
+			})
+		}
+		if image.UnprocessedImageRef.DigestRef == o.DigestRef() {
+			bundleProcessedImage = image
+		}
+	}
+
+	originBundleDigestRef, err := regname.NewDigest(o.DigestRef())
+	if err != nil {
+		return err
+	}
+	destinationRef, err := regname.ParseReference(bundleProcessedImage.DigestRef)
+	if err != nil {
+		return err
+	}
+
+	destDigestRef := destinationRef.Context().Digest(originBundleDigestRef.DigestStr())
+
```

Comment:
isnt ref that we care about here the bundleProcessedImage.DigestRef? why do we need to use digest from o.DigestRef? (two should be the same).

Reply:
yes you are right